src/mixins/child-apps.js

```diff
@@ -87,11 +99,11 @@ export default {
    * @method _stopChildApps
    */
   _stopChildApps() {
-    _.each(this._childApps, function(childApp) {
-      if(_.result(childApp, 'stopWithParent')) {
+    _.each(this._childApps, _.bind(function(childApp) {
+      if(_.result(childApp, this._isRestarting ? 'restartWithParent' : 'stopWithParent')) {
```

Comment:
the `_isRestarting` check could be done outside of the `_.each` loop.  This'd make it only be checked once and the bind wouldn't be necessary.